pkg/controller/backup_deletion_controller.go

```diff
@@ -54,250 +49,206 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const resticTimeout = time.Minute
-
-type backupDeletionController struct {
-	*genericController
-
-	deleteBackupRequestClient velerov1client.DeleteBackupRequestsGetter
-	deleteBackupRequestLister velerov1listers.DeleteBackupRequestLister
-	backupClient              velerov1client.BackupsGetter
-	restoreLister             velerov1listers.RestoreLister
-	restoreClient             velerov1client.RestoresGetter
-	backupTracker             BackupTracker
-	resticMgr                 restic.RepositoryManager
-	podvolumeBackupLister     velerov1listers.PodVolumeBackupLister
-	kbClient                  client.Client
-	snapshotLocationLister    velerov1listers.VolumeSnapshotLocationLister
-	csiSnapshotLister         snapshotv1listers.VolumeSnapshotLister
-	csiSnapshotContentLister  snapshotv1listers.VolumeSnapshotContentLister
-	csiSnapshotClient         *snapshotterClientSet.Clientset
-	processRequestFunc        func(*velerov1api.DeleteBackupRequest) error
-	clock                     clock.Clock
-	newPluginManager          func(logrus.FieldLogger) clientmgmt.Manager
-	backupStoreGetter         persistence.ObjectBackupStoreGetter
-	metrics                   *metrics.ServerMetrics
-	helper                    discovery.Helper
+const (
+	resticTimeout             = time.Minute
+	deleteBackupRequestMaxAge = 24 * time.Hour
+)
+
+type backupDeletionReconciler struct {
+	client.Client
+	logger            logrus.FieldLogger
+	backupTracker     BackupTracker
+	resticMgr         restic.RepositoryManager
+	metrics           *metrics.ServerMetrics
+	clock             clock.Clock
+	discoveryHelper   discovery.Helper
+	newPluginManager  func(logrus.FieldLogger) clientmgmt.Manager
+	backupStoreGetter persistence.ObjectBackupStoreGetter
+
+	podvolumeBackupLister velerov1listers.PodVolumeBackupLister
```

Comment:
Why are we still using a lister and not using the built-in controller-runtime cache?

Reply:
I used to think it's needed by 
`func GetSnapshotsInBackup` in restic/common.go which may be called in other places.

Just double checked this func is only used by this controller, I'll make the update, thanks!

Reply:
I've made the change in restic/common.go to remove the dependency of lister.